src/components/TestRunList.tsx

```diff
@@ -56,8 +59,61 @@ const TestRunList: React.FunctionComponent<{
     setDeleteDialogOpen(!deleteDialogOpen);
   };
 
+  const doesHaveUnapprovedImage = (): boolean => {
+    let status = false;
+    items.forEach((eachItem) => {
+      if (eachItem.status === TestStatus.new || eachItem.status === TestStatus.unresolved) {
+        status = true;
+        return;
+      }
+    });
+    return status;
+  };
+
+  const approveAll = () => {
+    enqueueSnackbar("Wait for the confirmation message when approval is completed.", {
+      variant: "info",
+    });
+    let lastItemToModify: TestRun | null = null;
+    //Determine the last item to be modified.
```

Comment:
could you wrap all async request with `await Promise.all` construction 
this gonna help to determine when all requests are completed

```js
await Promise.all(
      items
        .filter(isUnapproved)
        .map((item) =>
          testRunService
            .approve(item.id, false)
            .then((testRun) => updateTestRun(testRunDispatch, testRun))
        )
    )
      .then(() =>
        enqueueSnackbar("All approved", {
          variant: "success",
        })
      )
      .catch((err) =>
        enqueueSnackbar(err, {
          variant: "error",
        })
      );
```